Services/DeviceModels.cs

```diff
@@ -95,5 +97,54 @@ private List<string> GetDeviceModelFiles()
 
             return this.deviceModelFiles;
         }
+
+        public HashSet<string> GetPropNames()
```

Comment:
code style: all public methods first, before the private methods

Reply:
done